### DISCUSSION TITLE
Code-annotation does not work with standard r chunks containing echo: fenced

### DISCUSSION THREAD

**Discussion by petzi53** [quarto-dev/quarto-cli#6225] [Q&A]
### Description

I am working on a quarto book project and wanted to annotate some code lines. But inserting an R code chunk in RStudio via the menu "Code->Insert Chunk" (CMD-ALT-I on macOS) did not work when the option `#| echo: fenced` was included in the chunk. 

````qmd

```{r}
#| label: test-code-annotation
#| echo: fenced
library(tidyverse)
library(palmerpenguins)
penguins |>                                      # <1>
  mutate(                                        # <2>
    bill_ratio = bill_depth_mm / bill_length_mm, # <2>
    bill_area  = bill_depth_mm * bill_length_mm  # <2>
  )                                              # <2>
```
1.  Take `penguins`, and then,
2.  add new columns for the bill ratio and bill area.
````  

- [x] It worked as in the example shown when `echo: fenced` is not included. 
- [x] It also worked when the curly brakes around `r`are removed.

This is not convenient, especially if people are using the global option to execute chunks with their options.

````qmd
execute:
    echo: fenced
````
Because in that case, you cannot use the RStudio menu or shortcut, and you have to change from visual to source mode.

***
Quarto version: 1.4.36
Operating system: MacOS Venture 13.4.1, MacBook Pro Intel
RStudio version: 2023.06.1+524

**Comment by cderv** [comment]
This is currently a limitation for code-annotation feature as `echo: fenced` block are not results code chunk, but source code chunk. 

Possible something we could support easily. I'll transform this to an issue. Thanks !

**Comment by mcanouil** [comment]
Closing as resolved (and in favour of https://github.com/quarto-dev/quarto-cli/issues/6244)
